pkg/provisioner/ironic/raid.go

```diff
@@ -29,6 +29,8 @@ func setTargetRAIDCfg(p *ironicProvisioner, ironicNode *nodes.Node, raid *metal3
 		p.log.Info("rootDeviceHints is used, the first volume of raid will not be set to root")
 	}
```

Comment:
I think here can be changed to the following:
```go
	// set root volume
	if hasRootDeviceHints  {
		 p.log.Info("rootDeviceHints is used, the first volume of raid will not be set to root")
	}
        logicalDisks[0].IsRootVolume = &hasRootDeviceHints 
```

Reply:
We'd have to invert the logic.